### ISSUE TITLE
PDP Riffmaster driver is stuck on WinUSB and now doesn't work on Clone Hero or Fortnite Festival

### ISSUE THREAD

**Issue by quandaledinglelingle** [TheNathannator/RB4InstrumentMapper#20]
I've switched my drivers to play both Fortnite Festival and Clone Hero many times before, but this issue started suddenly yesterday

In RB4 Instrument Mapper my riffmaster shows under WinUSB Xbox Controllers, and when I press Revert Driver it loads for a little bit then does nothing. ![image](https://github.com/TheNathannator/RB4InstrumentMapper/assets/172215186/5d10f6a6-26f8-434f-a8d4-2e88883df69f)

In my device manager I see PDP RiffMaster Guitar for Xbox under Universal Serial Bus devices. When I right click here and uninstall device, it just shows back up the exact same after restarting my PC.

I'm using the PDP Riffmaster for Xbox. I've updated to the latest version of RB4 Instrument Mapper and the issue is still happening. I've tried the controller on a friend's computer and it works perfectly.

**Comment by TheNathannator** [comment]
Yeah this happens on occasion, and I'm not sure why unfortunately. You can uninstall the driver manually, there are [instructions here](https://github.com/TheNathannator/RB4InstrumentMapper/blob/main/Docs/WinUSB/manual-winusb-install.md#remove-winusb).

I've been working on a new backend that should make this process unnecessary for Riffmasters, hopefully that'll work more reliably. Will be releasing a new version soon that includes it.

**Comment by smartpug967** [comment]
Same issue here. Difference is that whenever I try to uninstall the driver it reappears after I restart. I am on Windows 11 while it looks like the screenshots were on Windows 10, could that be the issue? Either way, is there any other way to reset my driver back to the default?

**Comment by TheNathannator** [comment]
Make sure you're ticking the `Delete the driver software for this device` checkbox if it exists:

![](https://github.com/TheNathannator/RB4InstrumentMapper/blob/main/Docs/Images/WinUSB/Uninstall/device-manager-delete-driver-software.png)

It can also just take multiple attempts sometimes. There's otherwise no real reason the driver should get stuck like that, and the method linked in my prior message is the *only* method to reset the drivers AFAIK.

**Comment by TheNathannator** [comment]
I suppose you could try hitting Update Driver and letting it automatically find the right driver, but I'm unclear as to whether the driver is reverting successfully in the first place and simply un-reverting when you restart your PC. The answer to that would change the troubleshooting process a bit.

**Comment by smartpug967** [comment]
Took a few attempts but it seems to work now (thank the heavens!). The driver is still listed in my device manager though?!? Just gonna play with my fingers crossed

**Comment by KoffeeLord** [comment]
Same here... 

On XBOX Series X it runs perfectly... on PC its kinda bricked with this Driver-Revert-Bug :-(

**Comment by TheNathannator** [comment]
Is the manual uninstallation process not working for you? There's unfortunately not much I can do if that's the case, that would be an issue on Windows' side.

Half-considering making it so you can't switch Riffmasters over to WinUSB using the built-in menu lol, it's proven to be far too problematic and is not the recommended way to use them regardless.

**Comment by EmpireDeadman** [comment]
is my xbox riffmaster dongle DOA? it wont even pair to the guitar and shows up as this in device manager

![Image](https://github.com/user-attachments/assets/16142fd8-6f5e-4c82-afb4-24653008e874)

**Comment by TheNathannator** [comment]
that's probably the worst state i've seen one in so far lol

Definitely try to get in contact with PDP support about that if you can, I'm not sure what might be causing that.

**Comment by polarjrLOL** [comment]
im having this same issue except after uninstalling using zadig it still no longer works and somehow after buying a new crkd guitar to replace my riffmaster because of this after the first use it has the same issue AND I DIDNT EVEN USE RB4INSTRUMENTMAPPER WITH IT
